src/kokkos/ekat_view_utils.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef EKAT_VIEW_UTILS_HPP
+#define EKAT_VIEW_UTILS_HPP
+
+#include "ekat_kokkos_meta.hpp"
+#include "ekat_type_traits.hpp"
+
+#include <Kokkos_Core.hpp>
+
+namespace ekat {
+
+template<typename DataTypeOut, typename DataTypeIn, typename... Props>
+typename std::enable_if<GetRanks<DataTypeOut>::rank_dynamic==0,
+                        Unmanaged<Kokkos::View<DataTypeOut,Props...>>>::type
+reshape (Kokkos::View<DataTypeIn,Props...> view_in) {
+  typename ekat::Unmanaged<Kokkos::View<DataTypeOut,Props...>> view_out(view_in.data());
+  assert (view_in.size()==view_out.size());
+  return view_out;
+}
+
+template<typename DataTypeOut, typename DataTypeIn, typename... Props>
+typename std::enable_if<GetRanks<DataTypeOut>::rank_dynamic==1,
+                        Unmanaged<Kokkos::View<DataTypeOut,Props...>>>::type
+reshape (Kokkos::View<DataTypeIn,Props...> view_in,
+         const int dim0) {
+  typename ekat::Unmanaged<Kokkos::View<DataTypeOut,Props...>> view_out(view_in.data(),dim0);
+  assert (view_in.size()==view_out.size());
+  return view_out;
+}
+
+template<typename DataTypeOut, typename DataTypeIn, typename... Props>
+typename std::enable_if<GetRanks<DataTypeOut>::rank_dynamic==2,
+                        Unmanaged<Kokkos::View<DataTypeOut,Props...>>>::type
+reshape (Kokkos::View<DataTypeIn,Props...> view_in,
+         const int dim0, const int dim1) {
+  typename ekat::Unmanaged<Kokkos::View<DataTypeOut,Props...>> view_out(view_in.data(),dim0,dim1);
+  assert (view_in.size()==view_out.size());
+  return view_out;
+}
+
+template <typename ViewType, typename rngAlg, typename PDF>
+typename std::enable_if<Kokkos::is_view<ViewType>::value, void>::type
+genRandArray(ViewType view, rngAlg &engine, PDF &&pdf) {
+  auto mirror = Kokkos::create_mirror_view(view);
+  for (int i = 0; i < mirror.size(); ++i) {
+    mirror.data()[i] = pdf(engine);
+  }
+  Kokkos::deep_copy(view, mirror);
+}
+
+template<typename ViewType>
+typename ViewType::HostMirror cmvdc (const ViewType& v)
```

Comment:
I think we had this declared a couple of times in unit tests, and we definitely do the same a few times in EAMxx. It seemed natural to just add it here.

Reply:
What do you think about e.g. `clone_mirror_view` instead of `cmvdc`?